### PR TITLE
Updated SPARC logic and 32/64bit logic for x86 and i386

### PR DESCRIPTION
**Pull request by scotthain**
Adds sun4v and sun4u to 64bit arch list, since any OS that can runs on those that can use chef-sugar will be 64bit. (not guaranteed - I'm sure there could be an extreme edge case, but highly unlikely and very unsupportable).

Breaks out i386? - this should be Intel specific.

Add i386 arch to intel? - this is a valid option. (Verified vs fauxhai data)

Also added tests to cover as many edge cases as possible.

### CODE REVIEW

File: lib/chef/sugar/architecture.rb [hunk 1]
```diff
@@ -47,7 +55,7 @@ def _32_bit?(node)
       # @return [Boolean]
       #
       def intel?(node)
-        %w(i86pc)
+        %w(i86pc i386)
```
Comment (schisamo) [inline lib/chef/sugar/architecture.rb]: I think there are some missing values here. Just off the top of my head `i686` but here are probably others.

Comment (sethvargo) [inline lib/chef/sugar/architecture.rb]: Ping @scotthain

**Comment by scotthain** [comment]
Thanks for the reminder @sethvargo!

**Comment by sethvargo** [comment]
@schisamo any final thoughts on this before I merge?

File: spec/unit/chef/sugar/architecture_spec.rb [hunk 2]
```diff
@@ -33,31 +34,43 @@
...
       it "returns false when the system is #{arch}" do
         node = { 'kernel' => { 'machine' => arch } }
+        expect(described_class.i386?(node)).to be false
+      end
+    end
+  end
+
+  describe '#intel?' do
+    _intel_machines.each do |arch|
+      it "returns false when the system is #{arch}" do
```
Comment (schisamo) [inline spec/unit/chef/sugar/architecture_spec.rb]: I think this should be it returns true right?
